rmw_opendds_cpp/src/rmw_subscription.cpp

```diff
@@ -111,62 +128,31 @@ rmw_create_subscription(
   const rmw_qos_profile_t * qos_profile,
   const rmw_subscription_options_t* subscription_options)
 {
-  if (!node) {
-    RMW_SET_ERROR_MSG("node is null");
-    return nullptr;
-  }
-  RMW_CHECK_TYPE_IDENTIFIERS_MATCH(
-    node handle, node->implementation_identifier, opendds_identifier,
-    return nullptr)
+  RMW_CHECK_FOR_NULL_WITH_MSG(node, "node is null", return nullptr);
+  RMW_CHECK_TYPE_IDENTIFIERS_MATCH(node, node->implementation_identifier, opendds_identifier, return nullptr)
 
   auto node_info = static_cast<OpenDDSNodeInfo *>(node->data);
-  if (!node_info) {
-    RMW_SET_ERROR_MSG("node info is null");
-    return nullptr;
-  }
-
-  auto participant = static_cast<DDS::DomainParticipant *>(node_info->participant);
-  if (!participant) {
-    RMW_SET_ERROR_MSG("participant is null");
-    return nullptr;
-  }
+  RMW_CHECK_FOR_NULL_WITH_MSG(node_info, "node_info is null", return nullptr);
+  RMW_CHECK_FOR_NULL_WITH_MSG(node_info->participant.in(), "participant is null", return nullptr);
+  RMW_CHECK_FOR_NULL_WITH_MSG(node_info->subscriber_listener, "subscriber_listener is null", return nullptr);
 
   // message type support
   const rosidl_message_type_support_t * type_support = rmw_get_message_type_support(type_supports);
   if (!type_support) {
     return nullptr;
   }
-  const message_type_support_callbacks_t * callbacks =
-    static_cast<const message_type_support_callbacks_t *>(type_support->data);
+  auto callbacks = static_cast<const message_type_support_callbacks_t*>(type_support->data);
   if (!callbacks) {
     RMW_SET_ERROR_MSG("callbacks handle is null");
     return nullptr;
   }
   std::string type_name = _create_type_name(callbacks, "msg");
   OpenDDSStaticSerializedDataTypeSupport_var ts = new OpenDDSStaticSerializedDataTypeSupportImpl();
-  if (ts->register_type(participant, type_name.c_str()) != DDS::RETCODE_OK) {
+  if (ts->register_type(node_info->participant.in(), type_name.c_str()) != DDS::RETCODE_OK) {
```

Comment:
```suggestion
  if (ts->register_type(node_info->participant, type_name.c_str()) != DDS::RETCODE_OK) {
```

Reply:
pass DDS::DomainParticipant_var to DDS::DomainParticipant_ptr?